src/main/java/com/laborete/LaboreteAPI/posts/service/PostsServiceImpl.java

```diff
@@ -0,0 +1,99 @@
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@Service
+public class PostsServiceImpl implements PostsService {
+    private static final String USER_NOT_FOUND = "User was not found with id:";
+    private static final String POST_TEXT_IS_REQUIRED = "Post cannot be without text";
+    private static final String USER_CREATOR_IS_REQUIRED = "Post cannot be posted without User";
+    private static final String UUID_REQUIRED = "Can`t find post without ID";
+    private static final String POST_NOT_FOUND = "Post was not found with id:";
+    @Autowired
+    private PostRepository postRepository;
+    @Autowired
+    private UsersService usersService;
+    @Autowired
+    private PostMapper postMapper;
+    @Autowired
+    private UserMapper userMapper;
+
+    public PostsServiceImpl() {
+    }
+
+    public List<PostDTO> getAllPosts() {
+        List<PostEntity> postEntities = this.postRepository.findAll();
+        List<PostDTO> postDTOList = new ArrayList();
+        for (PostEntity postEntity : postEntities) {
+            PostDTO postDTO = this.postMapper.postEntityToPostDTO(postEntity);
+            postDTO.setUserId(postEntity.getUser().getId());
+            postDTO.setUserFirstName(postEntity.getUser().getFirstName());
+            postDTO.setUserSecondName(postEntity.getUser().getSecondName());
+            postDTO.setUserPosition(postEntity.getUser().getPosition());
+            postDTO.setUserAvatarDTO(this.usersService.getUserById(postEntity.getUser().getId()).getUserAvatarEntity());
+            postDTOList.add(postDTO);
+        }
+
+        return postDTOList;
+    }
+
+    public PostDTO createPost(CreatePostDTO post) {
+        if (post.getText() == null || post.getText().isEmpty()) {
+            throw new ResourceBadRequestException(POST_TEXT_IS_REQUIRED);
+        }
+        if (post.getUserId() == null) {
+            throw new ResourceBadRequestException(USER_CREATOR_IS_REQUIRED);
+        }
+        UserEntity userEntity = this.userMapper.userDTOToUserEntity(this.usersService.getUserById(post.getUserId()));
+        UserDTO userDTO = this.usersService.getUserById(post.getUserId());
+        PostEntity postEntity = new PostEntity();
+        postEntity.setId(post.getId());
+        postEntity.setText(post.getText());
+        postEntity.setCreationDate(LocalDateTime.now());
+        postEntity.setUser(userEntity);
+        this.postRepository.save(postEntity);
+        PostDTO postDTO = this.postMapper.postEntityToPostDTO(postEntity);
+        postDTO.setUserId(userEntity.getId());
+        postDTO.setUserFirstName(userEntity.getFirstName());
+        postDTO.setUserSecondName(userEntity.getSecondName());
+        postDTO.setUserPosition(userEntity.getPosition());
+        postDTO.setUserAvatarDTO(userDTO.getUserAvatarEntity());
+        return postDTO;
+    }
+
+
+    public PostDTO findById(UUID id) {
```

Comment:
There was a good method naming convension. 
`getPostById` -> returns a post if it's found; otherwise throws.
`findPostById` -> returns a post if it's found; otherwise returns null (in java there is an alternative to this approach - `Optional`).